### PR TITLE
[DMS] add document type table + make path NOT NULL, identification can be NULL

### PR DESCRIPTION
**Pull request by 3nids**
see https://github.com/qwat/QWAT/pull/317 for context

### CODE REVIEW

File: value_lists/vl_document_type.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ALTER TABLE qwat_vl.document_type ADD COLUMN active     boolean default true; /* determines if the actual document_type is active or not */
+
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (101, 'other', 'autre', 'alta');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (102, 'unknown', 'inconnu', 'necunoscută');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (103, 'to be determined', 'à déterminer', 'de determinat');
+
+/* VALUES */
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8001, true, 'Picture', 'Picture');
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8002, true, 'PDF', 'PDF');
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8003, true, 'Other', 'Autre');
```
Comment (ponceta) [inline value_lists/vl_document_type.sql:22]: @3nids what is the use of having 103 and 8003 for type 'Other' ?

File: value_lists/vl_document_type.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+/* COLUMNS */
+ALTER TABLE qwat_vl.document_type ADD COLUMN active     boolean default true; /* determines if the actual document_type is active or not */
+
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (101, 'other', 'autre', 'alta');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (102, 'unknown', 'inconnu', 'necunoscută');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (103, 'to be determined', 'à déterminer', 'de determinat');
+
+/* VALUES */
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8001, true, 'Picture', 'Picture');
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8002, true, 'PDF', 'PDF');
```
Comment (ponceta) [inline value_lists/vl_document_type.sql:21]: I would not say that PDF is a document type.

If appliable I would better have it set to a doc type list like qgep :

https://user-images.githubusercontent.com/8614223/70708118-0b85fc80-1cda-11ea-966b-0c8456d310c9.png

If it is thought for a functionnality, I would rather use file extensions and not enter some manual attribute.

**Review by ponceta — CHANGES_REQUESTED** [review]
Small remark on document_types

Please feel free to call me tomorrow if you want to check on this!

File: value_lists/vl_document_type.sql (hunk 1 above)
Comment (3nids) [reply value_lists/vl_document_type.sql]: I removed the extra line

File: value_lists/vl_document_type.sql (hunk 2 above)
Comment (3nids) [reply value_lists/vl_document_type.sql]: I completed the list according to the QGEP list (without voice recording for now).

File: update/delta/delta_1.4.0_001_DMS.sql [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
 
+CREATE TABLE qwat_vl.document_type () INHERITS ( qwat_vl.value_list_base);
+ALTER TABLE qwat_vl.document_type ADD CONSTRAINT vl_document_type_pk PRIMARY KEY (id);
+COMMENT ON TABLE qwat_vl.document_type IS 'document_type table';
+ALTER TABLE qwat_vl.document_type ADD COLUMN active     boolean default true; /* determines if the actual document_type is active or not */
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (101, 'other', 'autre', 'alta');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (102, 'unknown', 'inconnu', 'necunoscută');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (103, 'to be determined', 'à déterminer', 'de determinat');
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8001, true, 'Picture', 'Image');
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8002, true, 'Movie', 'Vidéo');
```
Comment (ponceta) [inline update/delta/delta_1.4.0_001_DMS.sql:11]:
```suggestion
INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8002, true, 'Video', 'Vidéo');
```

File: value_lists/vl_document_type.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+/* COLUMNS */
+ALTER TABLE qwat_vl.document_type ADD COLUMN active     boolean default true; /* determines if the actual document_type is active or not */
+
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (101, 'other', 'autre', 'alta');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (102, 'unknown', 'inconnu', 'necunoscută');
+INSERT INTO qwat_vl.document_type (id, value_en, value_fr, value_ro ) VALUES (103, 'to be determined', 'à déterminer', 'de determinat');
+
+/* VALUES */
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8001, true, 'Picture', 'Image');
+INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8002, true, 'Movie', 'Vidéo');
```
Comment (ponceta) [inline value_lists/vl_document_type.sql:21]:
```suggestion
INSERT INTO qwat_vl.document_type (id, active, value_en, value_fr) VALUES (8002, true, 'Video', 'Vidéo');
```

**Review by ponceta — APPROVED** [review]
@3nids feel free to squash and merge! ;)

Thank you for the applied modifications.

**Pull request merged**